superset/assets/src/dashboard/util/publishSubscriberUtil.js

```diff
@@ -2,15 +2,61 @@ import * as _ from 'lodash';
 
 export const APPLY_FILTER = 'APPLY_FILTER';
 export const USE_AS_MODAL = 'USE_AS_MODAL';
-const CHART_UPDATE_SUCCEEDED = "CHART_UPDATE_SUCCEEDED"
+const CHART_UPDATE_SUCCEEDED = 'CHART_UPDATE_SUCCEEDED';
 //check slice is modal or not
 export function isModalSlice(slice) {
     let actions = getUniqueActionsForSlice(slice);
     return isUseAsModalActionExist(actions);
 }
 
 export function isUseAsModalActionExist(actions) {
-    return (actions && actions.indexOf(USE_AS_MODAL) != -1) ? true :false;
+    if (actions) {
+        if (actions.constructor == Object) {
+            return (actions[USE_AS_MODAL]) ? true : false;
+        }
+        else if (actions.constructor == Array) {
+            return (actions.indexOf(USE_AS_MODAL) != -1) ? true : false;
+        }
+    }
+    return false;
+}
+
+export function getActionsMapForSlice(slice) {
+    let actions = {};
+    if (slice && slice.formData && slice.formData.hasOwnProperty('subscriber_layers')) {
+        let subscriberLayers = slice.formData.subscriber_layers
+        if (subscriberLayers) {
+            subscriberLayers.forEach(element => {
+                if (isUseAsModalActionExist(element.actions)) {
+                    if (!actions['USE_AS_MODAL'])
+                        actions['USE_AS_MODAL'] = [];
+                    actions['USE_AS_MODAL'] = _.union(actions['USE_AS_MODAL'], [element.linked_slice[0]['publisher_id']]);
```

Comment:
how we are ensuring that linked_slice will have only single element?

Reply:
every subscriberLayer can subscribed only one linked slice